pkg/controller/mizar/mizar-node-controller.go

```diff
@@ -0,0 +1,293 @@
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+	c := &MizarNodeController{
+		kubeclientset:  kubeclientset,
+		informer:       informer,
+		informerSynced: informer.Informer().HasSynced,
+		lister:         informer.Lister(),
+		recorder:       recorder,
+		queue:          queue,
+		grpcHost:       grpcHost,
+	}
+	klog.Infof(c.logInfoMessage("Sending events to api server"))
+	informer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(object interface{}) {
+			resource := object.(*v1.Node)
+			key := c.genKey(resource)
+			c.Enqueue(key, EventType_Create)
+			klog.Infof(c.logMessage("Create Node: ", key))
+		},
+		UpdateFunc: func(oldObject, newObject interface{}) {
+			oldResource := oldObject.(*v1.Node)
+			newResource := newObject.(*v1.Node)
+			oldKey := c.genKey(oldResource)
+			newKey := c.genKey(newResource)
+			if oldKey == newKey {
+				klog.Infof(c.logMessage("No actual change in nodes, discarding: ", newKey))
+			} else {
+				eventType, err := c.determineEventType(oldKey, newKey)
+				if err != nil {
+					klog.Errorf(c.logMessage("Unexpected string in queue; discarding: ", newKey))
+				} else {
+					c.Enqueue(newKey, eventType)
+				}
+				klog.Infof(c.logMessage("Update Node: ", newKey))
+			}
+		},
+		DeleteFunc: func(object interface{}) {
+			resource := object.(*v1.Node)
+			key := c.genKey(resource)
+			c.Enqueue(key, EventType_Delete)
+			klog.Infof(c.logMessage("Delete Node: ", key))
+		},
+	})
+	return c, nil
+}
+
+// Run starts an asynchronous loop that detects events of cluster nodes.
+func (c *MizarNodeController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+	klog.Infof(c.logInfoMessage("Starting node controller"))
+	klog.Infof(c.logInfoMessage("Waiting cache to be synced"))
+	if ok := cache.WaitForCacheSync(stopCh, c.informerSynced); !ok {
+		klog.Fatalf(c.logInfoMessage("Timeout expired during waiting for caches to sync."))
+	}
+	klog.Infof(c.logInfoMessage("Starting workers..."))
+	for i := 0; i < workers; i++ {
+		go wait.Until(c.runWorker, time.Second, stopCh)
+	}
+	<-stopCh
+	klog.Infof(c.logInfoMessage("Shutting down node controller"))
+}
+
+// Enqueue puts key of the node object in the work queue
+// EventType: Create=0, Update=1, Delete=2, Resume=3
+func (c *MizarNodeController) Enqueue(key string, eventType EventType) {
+	c.queue.Add(KeyWithEventType{Key: key, EventType: eventType})
+}
+
+// Dequeue an item and process it
+func (c *MizarNodeController) runWorker() {
+	for {
+		item, queueIsEmpty := c.queue.Get()
+		if queueIsEmpty {
+			break
+		}
+		c.process(item)
+	}
+}
+
+// Parsing a item key and call gRPC request
+func (c *MizarNodeController) process(item interface{}) {
+	defer c.queue.Done(item)
+	keyWithEventType, ok := item.(KeyWithEventType)
+	if !ok {
+		klog.Errorf(c.logMessage("Unexpected item in queue: ", keyWithEventType))
```

Comment:
Klog.Errorf supports formatting objects, so no need to to use c.logMessage to format the string.  Change line 150 to 

klog.Errorf"Unexpected item in queue: %v", keyWithEventType)

Reply:
c.logMessage() is removed and only klog() is using now.